client/src/main/java/io/split/client/SplitClientImpl.java

```diff
@@ -65,6 +65,11 @@ public String getTreatment(Key key, String split, Map<String, Object> attributes
             return Treatments.CONTROL;
         }
 
+        if (key.matchingKey() == null || key.bucketingKey() == null) {
```

Comment:
I'm not sure we should return control in this case. I believe people can still call us with `Key` and only fill up matching key. thoughts?